lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -286,11 +286,10 @@ void Visit(ParseNode *pnode, ByteCodeGenerator* byteCodeGenerator, PrefixFn pref
     }
     case knopClassDecl:
     {
-        // Visit the extends expression first, since it's bound outside the scope containing the class name.
-        Visit(pnode->sxClass.pnodeExtends, byteCodeGenerator, prefix, postfix);
         Visit(pnode->sxClass.pnodeDeclName, byteCodeGenerator, prefix, postfix);
         // Now visit the class name and methods.
         BeginVisitBlock(pnode->sxClass.pnodeBlock, byteCodeGenerator);
+        Visit(pnode->sxClass.pnodeExtends, byteCodeGenerator, prefix, postfix);
```

Comment:
I'll add a comment, but I think that comment was there because it was unusual/unexpected that the extends expression was bound outside the scope containing the class name. It seems that was true in the spec at some point but it was updated to the current behavior which has the extends clause within the scope containing the class name (where inside the extends clause the class name is an unbound reference)